docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
I do not think we should provide necessarily a get_public(), because you can already get the public_key out of the private key if you have pre-created it.
For ECDHE() we can simply have a private_key property which is the private_key that ECDHE() generated for you if you specify no private_key, then all you need to do to get the public_key() to send to the peer is:
ecdhe = ECDHE(ec.SECP256R1(), default_backend())
public_key = ecdhe.key.public_key()
shared_key = ecdhe.agree(peer_public_key, KDF())

I would like to keep the word agree() instead of get_shared as it convey the idea that a computation is being performed. Aslo whether the shared_key is really shared depends on the fact the correct KDF was used, so it may not be shared after all if you messed upo something (Ie you tried to "agree" on a shared key but failed :)